Shared/Samples/Show device location using indoor positioning/ShowDeviceLocationUsingIndoorPositioningView.Model.swift

```diff
@@ -0,0 +1,234 @@
+        
+        /// The location manager which handles the location data.
+        private let locationManager = CLLocationManager()
+        
+        /// The measurement formatter for sensor accuracy.
+        private let measurementFormatter: MeasurementFormatter = {
+            let formatter = MeasurementFormatter()
+            formatter.unitStyle = .short
+            formatter.unitOptions = .providedUnit
+            formatter.numberFormatter.maximumFractionDigits = 1
+            formatter.numberFormatter.roundingMode = .up
+            return formatter
+        }()
+        
+        /// This is the published value of the data that is displayed.
+        @Published private(set) var labelTextLeading: String = ""
+        
+        /// This is the published value of the data that is displayed.
+        @Published private(set) var labelTextTrailing: String = ""
+        
+        /// Represents loading state of indoors data, blocks interaction until loaded.
+        @Published var isLoading = false
+        
+        /// Kicks off the logic loading the data for the indoors map and indoors location.
+        func loadIndoorData() async throws {
+            isLoading = true 
+            try await map.load()
+            try await setIndoorDatasource()
+            try await requestLocationServicesAuthorizationIfNecessary()
+        }
+        
+        /// Stops the location data source.
+        func stopLocationDataSource() {
+            Task {
+                await locationDisplay.dataSource.stop()
+            }
+        }
+        
+        /// A function that attempts to load an indoor definition attached to the map
+        /// and returns a boolean value based whether it is loaded.
+        /// - Parameter map: The map that contains the IndoorDefinition.
+        /// - Returns: A boolean value for whether the IndoorDefinition is loaded.
+        private func indoorDefinitionIsLoaded(map: Map) async throws -> Bool {
+            guard map.indoorPositioningDefinition?.loadStatus != .loaded else { return true }
+            try await map.indoorPositioningDefinition?.load()
+            return map.indoorPositioningDefinition?.loadStatus == .loaded
+        }
+        
+        /// Sets the indoor datasource on the location display depending on
+        /// whether the map contains an IndoorDefinition.
+        private func setIndoorDatasource() async throws {
+            try await map.floorManager?.load()
+            // If an indoor definition exists in the map, it gets loaded and sets the IndoorsDataSource to pull information
+            // from the definition.
+            if try await indoorDefinitionIsLoaded(map: map),
+               let indoorPositioningDefinition = map.indoorPositioningDefinition {
+                indoorsLocationDataSource = IndoorsLocationDataSource(definition: indoorPositioningDefinition)
+                // Otherwise the IndoorsDataSource attempts to create itself using IPS table information.
```

Comment:
When commenting in `if-else` conditions, we typically put it in their respective scope. See `DownloadPreplannedMapAreaView.Model` line 102 for an example.